homeassistant/components/unifi/sensor.py

```diff
@@ -279,6 +283,75 @@ def make_wan_latency_entity_description(
     )
 
 
+@callback
+def async_device_temperatures_value_fn(
+    temperature_name: str, hub: UnifiHub, device: Device
+) -> float | None:
+    """Retrieve the temperature of the device."""
+    temperature = _device_temperature(temperature_name=temperature_name, device=device)
+
+    if TYPE_CHECKING:
+        # Checked by async_device_wan_latency_supported_fn
+        assert temperature
+
+    return temperature.get("value", 0)
+
+
+@callback
+def async_device_temperatures_supported_fn(
+    temperature_name: str, hub: UnifiHub, obj_id: str
+) -> bool:
+    """Determine if an device have a temperatures."""
+    if device := hub.api.devices[obj_id]:
+        return _device_temperature(temperature_name, device) is not None
+    return False
+
+
+@callback
+def _device_temperature(
+    temperature_name: str, device: Device
+) -> TypedDeviceTemperature | None:
+    """Return the temperature of the device."""
+    if device.temperatures:
+        for device_temperature in device.temperatures:
+            if temperature_name in device_temperature["name"]:
+                return device_temperature
+    return None
+
+
+def make_device_temperatur_sensors() -> tuple[UnifiSensorEntityDescription, ...]:
+    """Create device temperature sensors."""
+
+    def make_device_temperature_entity_description(
```

Comment:
See the changes to the sensor platform here https://github.com/home-assistant/core/pull/122875/files

Reply:
I think I've applied similar changes now